[OCPCLOUD-1436] AWS IMDSv2 support

lobziik opened the pull request:
enhancement: https://github.com/openshift/enhancements/pull/1075

Some additional changes in this PR:
- break dependency to cluster-api-aws, due to machine types was moved to openshift/api
- update govmomi due to go 1.16 dependency resolution issues

patrickdillon commented:
@lobziik are you hoping to land this for 4.11?

lobziik commented:
@patrickdillon yes...

patrickdillon commented on pkg/types/aws/machinepool.go:
```diff
@@ -25,6 +25,11 @@ type MachinePool struct {
+	// EC2MetadataOptions defines metadata service interaction options for EC2 instances in the machine pool.
+	//
+	// +optional
+	EC2Metadata `json:"metadataService"`
```
Don't think we should embed this:
```suggestion
	EC2Metadata EC2Metadata `json:"metadataService"`
```

patrickdillon commented on pkg/types/aws/validation/machinepool.go:
```diff
@@ -42,6 +44,11 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 	}
+
+	if p.EC2Metadata.Authentication != "" && !validMetadataAuthValues.Has(p.EC2Metadata.Authentication) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("authentication"), p.EC2Metadata.Authentication, "Metadata authentication must be either Required or Optional"))
```
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath.Child("authentication"), p.EC2Metadata.Authentication,  "must be either Required or Optional"))
```
The field name will already be present in the error message

lobziik commented on hack/tf-fmt.sh:
```diff
@@ -10,7 +10,7 @@ if [ "$IS_CONTAINER" != "" ]; then
   podman run --rm \
     --env IS_CONTAINER=TRUE \
-    --volume "${PWD}:${PWD}:z" \
+    --volume "${PWD}:${PWD}" \
```
Oops, committed that mistakenly. Will remove.
Podman on mac does not support this yet :(

lobziik commented on data/data/aws/cluster/master/main.tf:
```diff
@@ -146,6 +146,11 @@ resource "aws_instance" "master" {
   }
 
+  metadata_options {
+    http_endpoint = "enabled"
```
I had an issues during installation without this line.

lobziik commented:
Updated, moved openshift/api update to the first commit along with other vendoring.
There are some code which imports changing just for keep this revision working.

lobziik commented:
/retest

patrickdillon commented:
/lgtm
/approve
/refresh
/skip

patrickdillon commented:
/override ci/prow/e2e-metal-ipi-ovn-ipv6

patrickdillon commented:
/retest

patrickdillon commented:
/refresh

patrickdillon commented:
/override ci/prow/okd-images

This PR is not breaking okd images, which is having some sort of infrastructure issue ATM.
